Common/Orders/Fills/IFillModel.cs

```diff
@@ -13,61 +13,799 @@
  * limitations under the License.
 */
 
+using System;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using QuantConnect.Data;
+using QuantConnect.Data.Market;
+using QuantConnect.Interfaces;
 using QuantConnect.Securities;
 
 namespace QuantConnect.Orders.Fills
 {
     /// <summary>
-    /// Represents a model that simulates order fill events
+    /// Represents a base model that simulates order fill events
     /// </summary>
-    public interface IFillModel
+    public abstract class IFillModel : InterfaceFillModel
```

Comment:
cool, little switch-a-roo -- let's leave good documentation so future devs know _why_ it'slike this, because it does look strange, but there's good reasoning.